text/0008-bounded-contexts.md

```diff
@@ -0,0 +1,26 @@
+* Start Date: 11/13/2023
+* Contributors:
+  * [Craig McNally](cmcnally@ebsco.com)
+  * [Vince Bareau](vbareau@ebsco.com)
+* RFC PRs:
+  * PRELIMINARY REVIEW: https://github.com/folio-org/rfcs/pull/20
+  * DRAFT REFINEMENT: TBD
+  * PUBLIC REVIEW: TBD
+  * FINAL REVIEW: TBD
+* Outcome: 
+
+# Bounded contexts
+
+## Summary
+The term bounded context often appears in discussions about decomposing a monolithic system into multiple microservices.  Instead of all services sharing a single database, the system is partitioned into several bounded contexts, each with it's own database, data models, etc.  In Folio's case, we're working the other direction.  When the project started, it was decided that each module should have their own database, and cross-module database access must be avoided.  Coupling this with another practice the project has followed, separating business logic and storage modules, we wind up with suboptimal service/storage interactions.  It is proposed that we introduce the concept of bounded contexts, which would align with Application boundaries.  Any of the backend modules in a given Application would be able to directly access the storage layer.
+
+## Scope
+* Cross-module database access
+  * Data "ownership" and read vs write access
```

Comment:
We discussed at today's TC meeting that there is another RFC pending from September about mod-fqm-manager having permissions to ignore database boundaries.  That overlaps slightly then with this RFC in cross-module database access.  At the time there was disagreement about whether that permission would be specific to mod-fqm-manager or a more general exception, but @vbar-ebsco confirmed today he intends that RFC to be specific to mod-fqm-manager.  Therefore I think the overlap is minimal, and we can resolve it with a simple 'out-of-scope' statement here.   @craigmcnally said today he would add that statement.

As a straw man: "Access to any databases **outside** of the bounded context / application is out of scope, to be dealt with by a pending RFC about mod-fqm-manager".

Reply:
Thank you for the suggestion @maccabeelevine.  I've added a bullet to the scope statement.